tests/test_cache.py

```diff
@@ -62,7 +63,8 @@ def setUp(self):
         self.test_value = [1, 2, 3]
 
     def test_get_set(self):
-        t = tempfile.NamedTemporaryFile()
+        t = tempfile.NamedTemporaryFile(delete=False)
```

Comment:
Probably we can move this with required `self.addCleanup` calls to a helper function?

Reply:
generally imo low level two liner helpers are better inlining then forcing users to do the de-ref to understand the logic/flow especially if they have non obvious options set in this case delete=False, we do have some higher level helpers around tempfile/tempdir in c7n/testing.py.  the exception applies here though its commonly repeated, a helper local to this test class would simplify.

i'm actually a little unclear why this option (delete=False) is set in this usage as well. ideal usage here would be a context manager `with closing(tempfile.NamedTemporaryFile()) as t:`  afaics

Reply:
Delete pretty much universally needs to be false as the scenarios where it can be True and work on windows are few.

In this case as soon as you call cache.load it will fail if it was opened with `delete=True` as that will set the `O_Temporary` flag which will then be required for all future opens.   In Python 3 you can provide a custom opener to set that flag and we'd get the first `cache.load()` call to succeed, however when that load method finished the file would be deleted and the second `cache.load()` call would fail.  

I did make up a helper with a context manager to so you could do `with` and we'd `unlink` afterwards, but since these tests were all already using `add_cleanup` and the cleanup is significantly more reliable, I went ahead and left them that way.

I do think a little helper in this file would make it all more readable though, especially now that we have two cleanups on each.... I'm going to do that.